test/metabase/api/database_test.clj

```diff
@@ -376,44 +409,41 @@
                                                      :prefix "a"
                                                      :substring "a")))))))))))
 
-
-(defn- card-with-native-query {:style/indent 1} [card-name & {:as kvs}]
-  (merge
-   {:name          card-name
-    :database_id   (mt/id)
-    :dataset_query {:database (mt/id)
-                    :type     :native
-                    :native   {:query (format "SELECT * FROM VENUES")}}}
-   kvs))
-
-(defn- card-with-mbql-query {:style/indent 1} [card-name & {:as inner-query-clauses}]
-  {:name          card-name
-   :database_id   (mt/id)
-   :dataset_query {:database (mt/id)
-                   :type     :query
-                   :query    inner-query-clauses}})
-
-(defn- virtual-table-for-card [card & {:as kvs}]
-  (merge
-   {:id               (format "card__%d" (u/the-id card))
-    :db_id            (:database_id card)
-    :display_name     (:name card)
-    :schema           "Everything else"
-    :moderated_status nil
-    :description      nil}
-   kvs))
+(deftest card-autocomplete-suggestions-test
+  (let [result (fn [card] (select-keys card [:id :name :dataset]))]
+    (testing "GET /api/database/:id/card_autocomplete_suggestions"
+      (mt/with-temp* [Card [card-1 (card-with-native-query "Kanye West Quote Views Per Month")]
+                      Card [card-2 (card-with-native-query "Kanye West Quote Views Per Day")]]
+        (testing "exclude cards without perms"
+          (mt/with-non-admin-groups-no-root-collection-perms
+            (api.card-test/with-cards-in-readable-collection [card-1]
+              (is (= (->> [card-1] (map result))
```

Comment:
```suggestion
              (is (=  (map result [card-1])
```

if we just map through one function then I think threading is not needed.

Same to others test.

Reply:
Isn't this just `[(result card-1)]` with no `map` at all?

Reply:
Admittedly I was being very sloppy here. This should tidy things up https://github.com/metabase/metabase/commit/7c2005290871fd0bc70216ab0105a6b8a41d95a2